addons/delivery/models/sale_order.py

```diff
@@ -105,15 +105,29 @@ def _create_delivery_line(self, carrier, price_unit):
             'product_uom_qty': 1,
             'product_uom': carrier.product_id.uom_id.id,
             'product_id': carrier.product_id.id,
-            'price_unit': price_unit,
             'tax_id': [(6, 0, taxes_ids)],
             'is_delivery': True,
         }
+        if carrier.invoice_policy == 'real':
```

Comment:
Care it could fait if there is no delivery line but the picking is confirmed with a carrier. In this case it could add a 0 line with estimated cost. However the picking is already shipped.